PixaView/Controller/TableFilterVC.swift

```diff
@@ -0,0 +1,179 @@
+    func filterButtonDidTap(filterCategory: [Categories])
+}
+
+class TableFilterVC: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    
+    //MARK:  Properties-
+    let tableView = UITableView()
+    let filterCategory:[Categories] = Categories.allCases
+    var holder: PixaViewController?
+    weak var tableViewFilterSelectedDelegate: TableFilterViewControllerDelegate?
+    var selectedCategories: [Categories] = []
+    //MARK: - end of Properties
+    
+    //MARK:  viewDidLoad -
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.addSubview(tableView)
+        title = "Pixa View Filters"
+        
+        //cancel button in nav bar
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: "Cancel",
+                                                           style: .plain,
+                                                           target: self,
+                                                           action: #selector(cancelButtonFilterVC))
+        //search button in nav bar
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Search",
+                                                            style: .plain,
+                                                            target: self,
+                                                            action: #selector(searchButtonFilterVC))
+        
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
+        tableView.register(TableHeader.self, forHeaderFooterViewReuseIdentifier: "header")
+        
+        tableView.delegate = self
+        tableView.dataSource = self
+        
+        
+    }//end of viewDidLoad
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        tableView.frame = view.bounds
+    }
+
+    
+    //MARK: button actions-
+    
+    //cancel button nav bar method
+    @objc private func cancelButtonFilterVC(){
+        dismiss(animated: true, completion: nil)
+    }
+    
+    //search button nav bar method
+    @objc private func  searchButtonFilterVC(){
+        self.navigationController?.popViewController(animated: true)
+        
+        tableViewFilterSelectedDelegate?.filterButtonDidTap(filterCategory: selectedCategories)
+        
+        dismiss(animated: true, completion: nil)
+        
+    }
+    //MARK: - end of button actions
+    
+    //MARK: tаble methods -
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return filterCategory.count
+    }
+    
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
+        cell.textLabel?.text = filterCategory[indexPath.row].rawValue.capitalized
+
+        if selectedCategories.contains(filterCategory[indexPath.row]){
+            cell.accessoryType = .checkmark
```

Comment:
You can simplify it with

`cell.accessoryType = selectedCategories.contains(filterCategory[indexPath.row]) ?  .checkmark : .none`